src/device/input.cpp

```diff
@@ -6,21 +9,47 @@ auto device::InputModules::read_joy() -> std::pair<float, float> {
 
 auto device::InputModules::read_gyro() -> std::array<float, 3> {
     std::array<float, 3> gyro;
-    this->mpu.getGyro(gyro.data());
+    this->mpu->getGyro(gyro.data());
     return gyro;
 }
 
-// NOLINTBEGIN(bugprone-easily-swappable-parameters)
-device::InputModules::InputModules(
-    const std::pair<PinName, PinName>& joy_pins, const PinName& volume_pin,
-    const std::pair<PinName, PinName>& mpu_pins) :
-    joy(joy_pins.first, joy_pins.second),
-    volume(volume_pin),
-    mpu(mpu_pins.first, mpu_pins.second) {}
-// NOLINTEND(bugprone-easily-swappable-parameters)
+auto device::InputModules::Builder::joy_x_pin(const PinName& pin) -> Builder& {
+    this->_joy_x_pin = pin;
+    return *this;
+}
+
+auto device::InputModules::Builder::joy_y_pin(const PinName& pin) -> Builder& {
+    this->_joy_y_pin = pin;
+    return *this;
+}
+
+auto device::InputModules::Builder::volume_pin(const PinName& pin) -> Builder& {
+    this->_volume_pin = pin;
+    return *this;
+}
+
+auto device::InputModules::Builder::mpu_sda_pin(const PinName& pin) -> Builder& {
+    this->_mpu_sda_pin = pin;
+    return *this;
+}
+
+auto device::InputModules::Builder::mpu_scl_pin(const PinName& pin) -> Builder& {
+    this->_mpu_scl_pin = pin;
+    return *this;
+}
+
+auto device::InputModules::Builder::build() -> InputModules {
+    return InputModules{
+        std::make_pair<mbed::AnalogIn, mbed::AnalogIn>(this->_joy_x_pin, this->_joy_y_pin),
+        this->_volume_pin, std::make_unique<MPU6050>(this->_mpu_sda_pin, this->_mpu_scl_pin)};
+}
```

Comment:
なんか長いな...

```suggestion
auto device::InputModules::Builder::build() -> InputModules {
    const mbed::AnalogIn joy_x(this->_joy_x_pin);
    const mbed::AnalogIn joy_y(this->_joy_y_pin);
    const mbed::AnalogIn volume(this->_volume_pin);
    const auto mpu = std::make_unique<MPU6050>(this->_mpu_sda_pin, this->_mpu_scl_pin);
    return InputModules{std::make_pair(joy_x, joy_y), volume, mpu};
}
```

`const`はなくてもまあいいんだけど、可能であればつけておきたいです。あと`mpu`を渡すところに`std::move`つけないといけないかもです。